src/util/createServerRootMixin.js

```diff
@@ -104,63 +94,44 @@ function augmentInstantSearch(
     }
 
     let app;
-    let renderedComponent;
+    let instance;
 
     return Promise.resolve()
       .then(() => {
         app = cloneComponent(component, {
           mixins: [
             {
               created() {
-                // eslint-disable-next-line consistent-this
-                renderedComponent = this;
-                this.instantsearch.helper = helper;
-                this.instantsearch.mainHelper = helper;
+                instance = this.instantsearch;
 
-                this.instantsearch.mainIndex.init({
-                  instantSearchInstance: this.instantsearch,
-                  parent: null,
-                  uiState: this.instantsearch._initialUiState,
-                });
+                instance.start();
+                // although we use start for initializing the main index,
+                // we don't want to send search requests yet
+                instance.started = false;
               },
             },
           ],
         });
       })
       .then(() => renderToString(app))
-      .then(() => searchOnlyWithDerivedHelpers(helper))
+      .then(() => searchOnlyWithDerivedHelpers(instance.mainHelper))
       .then(() => {
-        const results = {};
-        walkIndex(renderedComponent.instantsearch.mainIndex, widget => {
-          results[widget.getIndexId()] = widget.getResults();
+        resultsState = {};
```

Comment:
We call this `initialResults` in InstantSearch now.

Reply:
6e69caf